torch/onnx/_internal/exporter.py

```diff
@@ -0,0 +1,248 @@
+# necessary to surface onnx.ModelProto through ExportOutput:
+from __future__ import annotations
+
+import abc
+import io
+import logging
+from typing import (
+    Any,
+    Callable,
+    Final,
+    Mapping,
+    Optional,
+    Protocol,
+    runtime_checkable,
+    Sequence,
+    TYPE_CHECKING,
+    TypeVar,
+    Union,
+)
+
+import torch
+
+from torch.onnx._internal import _beartype
+
+# We can only import onnx from this module in a type-checking context to ensure that
+# 'import torch.onnx' continues to work without having 'onnx' installed. We fully
+# 'import onnx' inside of dynamo_export (by way of _assert_dependencies).
+if TYPE_CHECKING:
+    import onnx
+
+
+_DEFAULT_OPSET_VERSION: Final[int] = 18
+"""The default ONNX opset version the exporter will use if one is not specified explicitly
+through ``ExportOptions``. This should NEVER be accessed outside of this module! Users
+should reference ``ExportOptions.opset_version``."""
+
+
+@_beartype.beartype
```

Comment:
this needs to move to `__init__`, now that `ExportOptions` is no longer `dataclass`.

Reply:
Looks like we can still decorate a class: https://beartype.readthedocs.io/en/latest/api_decor/#class-mode